components/script/dom/gpudevice.rs

```diff
@@ -781,33 +846,37 @@ impl GPUDeviceMethods for GPUDevice {
             self.channel.clone(),
             self,
             encoder,
-            descriptor.parent.label.clone().unwrap_or_default(),
+            descriptor.parent.label.clone(),
         )
     }
 
     /// <https://gpuweb.github.io/gpuweb/#dom-gpudevice-createtexture>
     fn CreateTexture(&self, descriptor: &GPUTextureDescriptor) -> Fallible<DomRoot<GPUTexture>> {
+        // TODO(sagudev): This should be https://gpuweb.github.io/gpuweb/#abstract-opdef-validate-gpuextent3d-shape
         let size = convert_texture_size_to_dict(&descriptor.size);
-        let desc = wgt::TextureUsages::from_bits(descriptor.usage).map(|usg| {
-            wgpu_res::TextureDescriptor {
-                label: convert_label(&descriptor.parent),
-                size: convert_texture_size_to_wgt(&size),
-                mip_level_count: descriptor.mipLevelCount,
-                sample_count: descriptor.sampleCount,
-                dimension: match descriptor.dimension {
-                    GPUTextureDimension::_1d => wgt::TextureDimension::D1,
-                    GPUTextureDimension::_2d => wgt::TextureDimension::D2,
-                    GPUTextureDimension::_3d => wgt::TextureDimension::D3,
-                },
-                format: convert_texture_format(descriptor.format),
-                usage: usg,
-                view_formats: descriptor
-                    .viewFormats
-                    .iter()
-                    .map(|tf| convert_texture_format(*tf))
-                    .collect(),
-            }
-        });
+        // TODO(sagudev): We should pass invalid bits to wgpu
```

Comment:
should or should not?

Reply:
we should, because wgpu-core will do check for invalid bits and generate proper validation errors.